contracts/core/StakeInterface.sol

```diff
@@ -0,0 +1,151 @@
+pragma solidity ^0.4.23;
+
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** @title The interface for the stake contract on origin. */
+interface StakeInterface {
+
+    /**
+     * @notice The message sender deposits the given amount of OST in the stake
+     *         contract to give the validator addresses voting power equal to
+     *         the deposit. The validator will be able to cast votes starting
+     *         at the current OSTblock height plus two.
+     *         Prior to this call, the message sender must approve an OST
+     *         transfer of the specified deposit ammount from her account to
+     *         the stake contract.
+     *
+     * @param _amount The amount of OST to deposit.
+     * @param _originValidatorAddress The address of the validator on origin.
+     * @param _auxiliaryValidatorAddress The address of the validator on
+     *                                   auxiliary.
+     * @param _withdrawalAddress The address where the OST should be
+     *                           transferred in case of a withdrawal.
+     *
+     * @return The unique index of the registered validator.
+     */
+    function deposit(
+        uint256 _amount,
+        address _originValidatorAddress,
```

Comment:
`deposit(uint256 _amount, address _validator)`; `msg.sender` is the owner on origin Core to deposit and withdrawal for the validator address (votes are cast on auxiliary)